python/idsse_common/test/test_http_utils.py

```diff
@@ -0,0 +1,161 @@
+    url = '/data/'+EXAMPLE_PROD_DIR+'/temp.grib2.gz'
+    httpserver.expect_request(url).respond_with_data(bytes([0,1,2]), status=200,
+                                                     content_type="application/octet-stream")
+    path = f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}/temp.grib2.gz'
+    dest = '/tmp/temp.grib2.gz'
+
+    copy_success = http_utils.http_cp(path, dest)
+    assert copy_success
+
+def test_http_cp_fails(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_PROD_DIR+'/temp.grib2.gz'
+    httpserver.expect_request(url).respond_with_data(bytes([0, 1, 2]), status=404,
+                                                     content_type="application/octet-stream")
+    path = f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}/temp.grib2.gz'
+    dest = '/tmp/temp.grib2.gz'
+    copy_success = http_utils.http_cp(path, dest)
+    assert not copy_success
+
+
+def test_check_for_succeeds(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+
+    result = http_utils.check_for(EXAMPLE_ISSUE, EXAMPLE_VALID)
+    assert result is not None
+    assert result == (EXAMPLE_VALID, f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}{EXAMPLE_FILES[1]}')
+
+
+def test_check_for_does_not_find_valid(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT
+    httpserver.expect_request(url).respond_with_data('', content_type="text/plain")
+    unexpected_valid = datetime(1970, 10, 3, 23, tzinfo=UTC)
+    result = http_utils.check_for(EXAMPLE_ISSUE, unexpected_valid)
+    assert result is None
+
+
+def test_get_issues(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/' + EXAMPLE_ENDPOINT +'/'
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+    result = http_utils.get_issues(issue_start=EXAMPLE_ISSUE,
+                                   time_delta=timedelta(minutes=1))
+    assert len(result) == 1
+    assert result[0] == EXAMPLE_ISSUE
+
+
+def test_get_issues_with_same_start_stop(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT+'/'
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+    result = http_utils.get_issues(issue_start=EXAMPLE_ISSUE, issue_end=EXAMPLE_ISSUE, time_delta=timedelta(minutes=1))
+    assert len(result) == 1
+    assert result[0] == EXAMPLE_ISSUE
+
+def test_get_valids_all(http_utils: HttpUtils, httpserver: HTTPServer):
+    url = '/data/'+EXAMPLE_ENDPOINT+'/'
+    httpserver.expect_request(url).respond_with_data(EXAMPLE_RETURN, content_type="text/plain")
+    result = http_utils.get_valids(EXAMPLE_ISSUE)
+    assert len(result) == 0
+    #assert result[0] == (EXAMPLE_VALID, f'{EXAMPLE_URL}{EXAMPLE_PROD_DIR}{EXAMPLE_FILES[1]}')
+
+
+def test_get_valids_with_wildcards(http_utils_with_wild: HttpUtils, httpserver: HTTPServer):
```

Comment:
Given that this http_utils_with_wild.get_valids doesn't find anything, is this a meaningful test?

Reply:
True, I'll add a meaningful test...

Reply:
Again if the filepath format contained valid this should return something. If you want, I'd be okay if we dropped this test unless it is need for code coverage. If you do drop it make sure to remove http_utils_with_wild feature if it isn't used anywhere else.